Source/JavaScriptCore/bytecode/MetadataTable.cpp

```diff
@@ -61,16 +61,17 @@ MetadataTable::~MetadataTable()
 
 void MetadataTable::destroy(MetadataTable* table)
 {
-    RefPtr<UnlinkedMetadataTable> unlinkedMetadata = WTFMove(table->linkingData().unlinkedMetadata);
-
-    table->~MetadataTable();
-
     // FIXME: This check should really not be necessary, see https://webkit.org/b/272787
-    if (UNLIKELY(!unlinkedMetadata)) {
+    if (table->isDestroyed()) {
         ASSERT_NOT_REACHED();
         return;
     }
 
+    RefPtr<UnlinkedMetadataTable> unlinkedMetadata = WTFMove(table->linkingData().unlinkedMetadata);
+    ASSERT(table->isDestroyed());
```

Comment:
Is it correct?

Reply:
I think so, move the `unlinkedMetadata` should just replace that with a nullptr then it is `isDestroyed`.

Reply:
Yes, the WTFMove basically exchanges the Ref with a nullptr.